### PR TITLE
Updating project to newer libraries versions. Adapt to AS 3.x

### PR DESCRIPTION
**Pull request by rcd27**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -129,12 +128,11 @@ dependencies {
         transitive = true
     }
 
-    // https://mvnrepository.com/artifact/org.apache.commons/commons-math3
-    compile files('libs/commons-math3-3.6.1.jar')
+    implementation 'org.apache.commons:commons-math3:3.6.1'
```
Comment (emartynov) [inline app/build.gradle]: Is it big library? Why do we need it?

File: app/build.gradle [hunk 2]
```diff
@@ -44,7 +43,7 @@ android {
 
     buildTypes {
         debug {
-            minifyEnabled false
+            minifyEnabled true
```
Comment (emartynov) [inline app/build.gradle]: It is needed for security but I prefer to have it in different PR

File: app/src/main/java/org/glucosio/android/GlucosioApplication.java [hunk 3]
```diff
@@ -42,7 +42,7 @@
 
 import uk.co.chrisjenx.calligraphy.CalligraphyConfig;
 
-public class GlucosioApplication extends Application {
+public class GlucosioApplication extends MultiDexApplication {
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/GlucosioApplication.java]: Why do we need it here? Did we already hit 65k limit?

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Mon Mar 20 12:26:00 CET 2017
+#Mon Nov 27 09:42:36 MSK 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions-snapshots/gradle-4.4-20171031235950+0000-all.zip
```
Comment (emartynov) [inline gradle/wrapper/gradle-wrapper.properties]: Please no snapshots

**Review by emartynov — CHANGES_REQUESTED** [review]
Please react to comments

File: app/src/main/java/org/glucosio/android/GlucosioApplication.java (hunk 3 above)
Comment (rcd27) [reply app/src/main/java/org/glucosio/android/GlucosioApplication.java]: That was a merge mistake. No need in multidex with proguard actually.

File: app/build.gradle (hunk 1 above)
Comment (rcd27) [reply app/build.gradle]: Seems like we don't.

File: app/build.gradle (hunk 2 above)
Comment (rcd27) [reply app/build.gradle]: With `minifyEnabled false` we have ` Total methods in app-debug.apk: 57762 (88.14% used)`
And with `true`: `Total methods in app-debug.apk: 34668 (52.90% used)`

Comment (emartynov) [reply app/build.gradle]: I know, I know, but 58k is below 65k

Comment (emartynov) [reply app/build.gradle]: Usually, you can not just switch on proguard, it requires checking if app still working as it was before, if we need to adjust default settings, if we want to add more specific to our project, etc

**Comment by emartynov** [comment]
And we have failing unit tests now!

File: app/build.gradle (hunk 2 above)
Comment (rcd27) [reply app/build.gradle]: Ok, got it.

**Comment by rcd27** [comment]
Was HelloActivityTest working before?

File: app/build.gradle [hunk 5]
```diff
@@ -129,25 +130,22 @@ dependencies {
...
     testCompile "junit:junit:${junitVer}"
     testCompile "org.mockito:mockito-all:${mockitoVer}"
     testCompile "com.squareup.assertj:assertj-android:${androidAssertJVer}"
-    testCompile("org.robolectric:robolectric:${robolectricVer}") {
+    testImplementation("org.robolectric:robolectric:3.5.1") {
         exclude group: 'commons-logging', module: 'commons-logging'
         exclude group: 'org.apache.httpcomponents', module: 'httpclient'
     }
-    testCompile("org.robolectric:shadows-support-v4:${robolectricVer}") {
+    testImplementation("org.robolectric:shadows-support-v4:3.3.2") {
```
Comment (emartynov) [inline app/build.gradle:144]: If you started moving to new android gradle plugin then why not change it everywhere?

File: app/src/test/java/org/glucosio/android/fragment/AssistantFragmentTest.java [hunk 6]
```diff
@@ -3,10 +3,13 @@
 import org.glucosio.android.RobolectricTest;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.shadows.support.v4.SupportFragmentTestUtil;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+@RunWith(RobolectricTestRunner.class)
```
Comment (emartynov) [inline app/src/test/java/org/glucosio/android/fragment/AssistantFragmentTest.java]: This is not needed! Check the `RobolectricTest` class

**Review by emartynov — APPROVED** [review]

**Comment by emartynov** [comment]
There are still two more comments. Could you address them?

File: app/src/test/java/org/glucosio/android/fragment/AssistantFragmentTest.java (hunk 6 above)
Comment (rcd27) [reply app/src/test/java/org/glucosio/android/fragment/AssistantFragmentTest.java]: Omg, missed that it extends RobolectricTest.class

**Comment by emartynov** [comment]
Looks good! Please continue with the contribution!

I'm going to run test first to see if the app works and if so I will accept it!

**Review by ghost — COMMENTED** [review]
Looks good to me. Tests passed :-)

**Comment by emartynov** [comment]
Quite solid contribution! Thank you!

I'm merging!

**Pull request merged**